lib/error_report_notify.c

```diff
@@ -0,0 +1,157 @@
+// SPDX-License-Identifier: GPL-2.0
```

Comment:
Commit-message: I think it needs to be better motivated (and don't mention KCSAN, since that's not happening in this series).

I would also strengthen the motivation by copying bits from the design doc:
```
With the introduction of various production error-detection tools, such as
MTE-based KASAN and KFENCE, the need arises to efficiently notify user space
tools about kernel errors. Currently, no facility exists to notify user space
about a kernel error from such bug-detection tools. The problem is obviously
not restricted to the above bug detection tools, and applies to any error
reporting mechanism that does not panic the kernel; this series, however, will
only add support for KASAN and KFENCE reporting.

All such error reports appear in the kernel log. But, when such errors
occur, user space would normally need to read the entire kernel log and parse
the relevant errors. This is error prone and inefficient, as user space needs
to continuously monitor the kernel log and parse errors. On certain devices,
this is unfortunately not acceptable. Therefore, we need to revisit how reports
are propagated to user space.

The library added, error_report_notify (CONFIG_ERROR_REPORT_NOTIFY),
solves the above by using the error_report_start/error_report_end ...<rest of text>...
```

Minor:
- the config does not collect errors. The library does.

Reply:
Changed the title to "lib: add error_report_notify to collect debugging tools' reports" and picked the rest of your text.
Thanks!